src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -605,6 +614,20 @@ public function getDriver()
         return $this->driver;
     }
 
+    /**
+     * Flush the Flysystem cache.
+     *
+     * @return void
+     */
+    public function flushCache()
+    {
+        $adapter = $this->driver->getAdapter();
+
+        if ($adapter instanceof CachedAdapter) {
```

Comment:
Can `$driver->getAdapter()` ever return something different? If not, why the if condition? If yes, what will happen then? A silent failure that lead to believe that the cache was flushed, but in fact it wasn't?

Reply:
Only `CachedAdapter` classes have `getCache()` methods.

The alternative might be to throw an exception if it's not a cached adapter, but then you'd need to do a check in your own code. Might as well make things easier.